src/ripple/rpc/handlers/Manifest.cpp

```diff
@@ -0,0 +1,75 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2019 Dev Null Productions
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/basics/base64.h>
+
+namespace ripple {
+Json::Value doManifest (RPC::Context& context)
+{
+    auto& params = context.params;
+
+    if(!params.isMember(jss::public_key))
+        return RPC::missing_field_error (jss::public_key);
+
+    auto requested = params[jss::public_key].asString();
+
+    Json::Value ret;
+    ret[jss::requested] = requested;
+
+    auto pk = parseBase58<PublicKey>(TokenType::NodePublic, requested);
+    if(!pk)
+    {
+        RPC::inject_error(rpcINVALID_PARAMS, ret);
+        return ret;
+    }
+
+    // first attempt to use params as ephemeral key,
+    // if this lookup succeeds master key will be returned,
+    // else pk will just be returned and we will assume that
+    // is master key anyways
+    auto mk = context.app.validatorManifests().getMasterKey(*pk);
+
+    auto ek = context.app.validatorManifests().getSigningKey(mk);
+
+    // if ephemeral key not found, we don't have specified manifest
+    if(ek == mk)
+      return ret;
+
+    auto manifest = context.app.validatorManifests().getManifest(mk);
+    if(manifest)
+      ret[jss::manifest] = base64_encode((std::uint8_t const*)manifest->c_str(),
```

Comment:
Two style nits:
1) should be indented four spaces
2) we usually use `static_cast` over c-style casts

Reply:
@movitto (responding to your comment here https://github.com/ripple/rippled/pull/3197#issuecomment-572272128)

`static_cast` won't work - we need a `reinterpret_cast` so something like:
```
        ret[jss::manifest] = base64_encode(
            reinterpret_cast<std::uint8_t const*>(manifest->data()),
            manifest->length());
```

*However*, there's an overload of `base64_encode` that takes a `std::string`; I'd use that. So we can change this to:
```
        ret[jss::manifest] = base64_encode(*manifest);
```